### PR TITLE
[DX-835] Update tyk-stack docs

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link

<br>

[preview - self managed](https://deploy-preview-3614--tyk-docs.netlify.app/docs/nightly/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack) <br>

[preview - oss](https://deploy-preview-3614--tyk-docs.netlify.app/docs/nightly/tyk-oss/ce-helm-chart-new/) <br>

[preview - hybrid](https://deploy-preview-3614--tyk-docs.netlify.app/docs/nightly/tyk-cloud/environments-deployments/hybrid-gateways/#deploy-in-kubernetes-with-helm)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md [hunk 1]
```diff
@@ -52,13 +56,70 @@ Also, you can set the version of each component through `image.tag`. You could f
 * [Helm 3+](https://helm.sh/docs/intro/install/)
 * [Redis](https://tyk.io/docs/tyk-oss/ce-helm-chart/#recommended-via-bitnami-chart) should already be installed or accessible by the gateway. 
 
+## Quick Start
+Quick start using `tyk-stack` and Bitnami Redis and MongoDB chart
```
Comment (dcs3spp) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md]: Explain what the overall purpose of the quick start is, something like:

```suggestion
The following quick starts guide explains how to use the Tyk Stack Helm chart to configure a Dashboard that includes:
- Tyk Pump to write analytics to MongoDB and Prometheus
- Redis for key storage

At the end of this quickstart Tyk Dashboard should be accessible through service `dashboard-svc-tyk-tyk-dashboard` at port `3000`.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR looks LGTM Carol, I have pushed some commits to the branch with a minor grammar change. I have added comment / suggestion relating to quickstart introduction

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md [hunk 2]
```diff
@@ -52,31 +55,126 @@ Also, you can set the version of each component through `image.tag`. You could f
...
+  --set global.secrets.useSecretName=my-secrets \
+  --set global.redis.addrs="{tyk-redis-master.$NAMESPACE.svc:6379}" \
+  --set global.redis.passSecret.name=tyk-redis \
+  --set global.redis.passSecret.keyName=redis-password \
+  --set global.postgres.connectionStringSecret.name=postgres-secrets \
+  --set global.postgres.connectionStringSecret.keyName=postgresUrl
+```
+
+## Quick Start with MongoDB
+The following quick starts guide explains how to use the Tyk Stack Helm chart to configure a Dashboard that includes:
```
Comment (dcs3spp) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md]:
```suggestion
The following quick start guide explains how to use the Tyk Stack Helm chart to configure a Dashboard that includes:
```

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md [hunk 3]
```diff
@@ -52,31 +55,126 @@ Also, you can set the version of each component through `image.tag`. You could f
 * [Helm 3+](https://helm.sh/docs/intro/install/)
 * [Redis](https://tyk.io/docs/tyk-oss/ce-helm-chart/#recommended-via-bitnami-chart) should already be installed or accessible by the gateway. 
 
+## Quick Start with PostgreSQL
+The following quick starts guide explains how to use the Tyk Stack Helm chart to configure a Dashboard that includes:
```
Comment (dcs3spp) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md]:
```suggestion
The following quick start guide explains how to use the Tyk Stack Helm chart to configure a Dashboard that includes:
```

**Comment by dcs3spp** [comment]
@caroltyk Thanks for updating changes, is PR ready for release?

**Comment by carolwkc** [comment]
> @caroltyk Thanks for updating changes, is PR ready for release?

Hi @dcs3spp , I'll add more changes to other helm chart related pages to Sync up the helm release status. Will need to wait for Tyk Charts 1.2 release too.

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md [hunk 4]
```diff
@@ -472,108 +700,35 @@ Uptime Pump can be configured by setting `pump.uptimePumpBackend` in values.yaml
...
+    dashboard: ""
 ```
 
+### Tyk Bootstrap
+
+To enable bootstrapping, set `global.components.bootstrap` to `true`. It would run [tyk-k8s-bootstrap](https://github.com/TykTechnologies/tyk-k8s-bootstrap) to bootstrap `tyk-stack` and to create Kubernetes secrets that can be utilized in Tyk Operator and Tyk Enterprise Developer Portal.
+
+#### Bootstrapped Environments
+
+If Tyk is already bootstrapped, the application will bypass the creation of the Tyk Organization and Admin User, proceeding directly with the creation of Kubernetes Secrets.
```
Comment (komalsukhani) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md:744]: @buraksekili We need to disable dashboard bootstrapping to bypass creation of org and admin user right?

File: tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways.md [hunk 5]
```diff
@@ -342,6 +341,57 @@ helm install redis tyk-helm/simple-redis -n tyk
 
 The Tyk Helm Chart can connect to `simple-redis` in the same namespace by default. You do not need to set Redis address and password in `values.yaml`.
 
+### Protect Confidential Fields with Kubernetes Secrets
+
+In the `values.yaml` file, some fields are considered confidential, such as `APISecret`, connection strings, etc.
+Declaring values for such fields as plain text might not be desired for all use cases. Instead, for certain fields,
+Kubernetes secrets can be referenced, and Kubernetes by itself configures values based on the referred secret.
```
Comment (komalsukhani) [inline tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways.md]:
```suggestion
Kubernetes secrets can be referenced, and Helm Chart by itself configures values based on the referred secret.
```
Do you mean Helm chart here?

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md (hunk 4 above)
Comment (buraksekili) [reply tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md:744]: If the organisation exists, we do not run tyk dashboard bootstrapping (creation of organisation and admin user). we do not need to disable bootstrapping explicitly.

File: tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways.md (hunk 5 above)
Comment (carolwkc) [reply tyk-docs/content/tyk-cloud/environments-deployments/hybrid-gateways.md]: @komalsukhani please see updates here:

1. https://github.com/TykTechnologies/tyk-docs/pull/3614/commits/c88c2ba8e2865882578a88f00f655a45721a5625

2. https://github.com/TykTechnologies/tyk-docs/pull/3614/commits/16abe9e05ee7891bb9feedac8c1b8356cbc0bd6a

3. https://github.com/TykTechnologies/tyk-docs/pull/3614/commits/c1b87a96ac09271ea4d1a619d54f691e7434d1ed

File: tyk-docs/content/tyk-oss/ce-helm-chart-new.md [hunk 6]
```diff
@@ -17,16 +17,16 @@ Tyk is working to provide a new set of Helm charts, and will progressively roll
...
+| tyk-stack          | Tyk Self Managed (Single Data Center) | Stable              |
+| tyk-control-plane  | Tyk Self Managed (Distributed) Control Plane | Coming Soon     |
+| tyk-data-plane     | Tyk Self Managed (Distributed) Data Plane <br> Tyk Hybrid Data Plane | Stable              |
 
 To deploy Tyk OSS using the new Helm chart, please use [tyk-oss](https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-oss) chart.
 
 ## Tyk Open Source
 
-Tyk-oss provides the default deployment of the Tyk Open Source stack. It includes the Tyk Gateway, an open source Enterprise API Gateway, supporting REST, GraphQL, TCP and gRPC protocols; and Tyk Pump, an analytics purger that moves the data generated by your Tyk gateways to any back-end.
+`Tyk-oss` provides the default deployment of the Tyk Open Source stack. It includes the Tyk Gateway, an open source Enterprise API Gateway, supporting REST, GraphQL, TCP and gRPC protocols; and Tyk Pump, an analytics purger that moves the data generated by your Tyk gateways to any back-end.
```
Comment (buraksekili) [inline tyk-docs/content/tyk-oss/ce-helm-chart-new.md]: You can ignore the suggestion if it is not suitable
```suggestion
`tyk-oss` provides the default deployment of the Tyk Open Source stack. It includes the Tyk Gateway, an open source Enterprise API Gateway, supporting REST, GraphQL, TCP and gRPC protocols; and Tyk Pump, an analytics purger that moves the data generated by your Tyk gateways to any back-end.
```

**Review by dcs3spp — APPROVED** [review]

**Comment by dcs3spp** [comment]
@caroltyk I have approved. Awaiting approval from peer reviewers and will then merge and release when Tyk Charts released

**Review by komalsukhani — APPROVED** [review]

File: tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md [hunk 7]
```diff
@@ -50,33 +53,129 @@ Also, you can set the version of each component through `image.tag`. You could f
 
 * [Kuberentes 1.19+](https://kubernetes.io/docs/setup/)
```
Comment (buraksekili) [inline tyk-docs/content/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack.md]:
```suggestion
* [Kubernetes 1.19+](https://kubernetes.io/docs/setup/)
```

**Pull request merged**